### PR TITLE
Pub/Sub: fix all publisher examples

### PR DESCRIPTION
**Pull request by anguillanneuf**
- add `future.result()` to resolve futures to ensure publish success
- update publish with error handler example some more

### CODE REVIEW

File: pubsub/cloud-client/subscriber_test.py [hunk 1]
```diff
@@ -169,13 +169,15 @@ def test_update(subscriber_client, subscription, capsys):
...
+        future = publisher_client.publish(
             topic, data=data)
+        future.result()
 
 
 def _publish_messages_with_custom_attributes(publisher_client, topic):
     data = u'Test message'.encode('utf-8')
-    publisher_client.publish(topic, data=data, origin='python-sample')
+    future = publisher_client.publish(topic, data=data, origin='python-sample')
+    future.result()
```
Comment (hongalex) [inline pubsub/cloud-client/subscriber_test.py]: Same as above: do we want to print this?

File: pubsub/cloud-client/subscriber_test.py [hunk 2]
```diff
@@ -169,13 +169,15 @@ def test_update(subscriber_client, subscription, capsys):
 def _publish_messages(publisher_client, topic):
     for n in range(5):
         data = u'Message {}'.format(n).encode('utf-8')
-        publisher_client.publish(
+        future = publisher_client.publish(
             topic, data=data)
+        future.result()
```
Comment (hongalex) [inline pubsub/cloud-client/subscriber_test.py]: Do we care about printing the result of this?

File: pubsub/cloud-client/publisher_test.py [hunk 3]
```diff
@@ -111,11 +111,7 @@ def test_publish_with_batch_settings(topic, capsys):
 
 
 def test_publish_with_error_handler(topic, capsys):
-
-    with _make_sleep_patch():
```
Comment (hongalex) [inline pubsub/cloud-client/publisher_test.py]: Why is this being removed?

File: pubsub/cloud-client/subscriber_test.py (hunk 1 above)
Comment (anguillanneuf) [reply pubsub/cloud-client/subscriber_test.py]: No we don't.

File: pubsub/cloud-client/subscriber_test.py (hunk 2 above)
Comment (anguillanneuf) [reply pubsub/cloud-client/subscriber_test.py]: Same as above.

File: pubsub/cloud-client/publisher_test.py (hunk 3 above)
Comment (anguillanneuf) [reply pubsub/cloud-client/publisher_test.py]: We no longer wait indefinitely. We just wait for the dictionary `futures` to run dry now.

**Pull request merged**